src/Factories/ResultsRendererFactory.php

```diff
@@ -14,6 +14,28 @@ class ResultsRendererFactory
     const FORMAT_TEXT = 'text';
 
     /**
+     * @var JsonResultsRenderer
+     */
+    protected $jsonRenderer;
+
+    /**
+     * @var ConsoleResultsRenderer
+     */
+    protected $consoleRenderer;
+
+    /**
+     * @param JsonResultsRenderer    $jsonRenderer
+     * @param ConsoleResultsRenderer $consoleRenderer
+     */
+    public function __construct(
+        JsonResultsRenderer $jsonRenderer,
```

Comment:
Thanks I understand your point here but in this case I believe it is ok. This is a factory class so using `new` to create the classes (and return their output) is fine here in my opinion.

So if you could remove the changes to this class please.

Reply:
But then its not unit test, its more like integration testing. Test with json format, i think, i will test (assert output write method), but how to test text format? ConsoleResultsRenderer created Table instance which just echoes the result...

Reply:
Ah ok I see your point. Yeah that is true. I'm thinking perhaps I should change the factory so it just returns the object then. Anyways, I see your updated code and it looks great so I'll merge it. Thanks very much!